src/ripple/nodestore/impl/EncodedBlob.cpp

```diff
@@ -18,24 +18,37 @@
 //==============================================================================
 
 #include <ripple/nodestore/impl/EncodedBlob.h>
+#include <cassert>
 #include <cstring>
 
 namespace ripple {
 namespace NodeStore {
 
-void
-EncodedBlob::prepare(std::shared_ptr<NodeObject> const& object)
+EncodedBlob::EncodedBlob(std::shared_ptr<NodeObject> const& object)
+    : size_(object->getData().size() + 9)
+    , ptr_(
+          (size_ <= payload_.size()) ? payload_.data()
+                                     : new std::uint8_t[size_])
 {
-    m_key = object->getHash().begin();
+    std::memcpy(
+        key_.data(), object->getHash().data(), object->getHash().size());
 
-    auto ret = m_data.alloc(object->getData().size() + 9);
+    for (std::size_t i = 0; i != 8; ++i)
+        ptr_[i] = 0;
```

Comment:
I am curious why you don't use `std::memset()` here.

Reply:
No particular reason; I could have used `std::memset` or `std::fill` even. Generally, the compiler should optimize all these away (see [godbolt](https://godbolt.org/#g:!((g:!((g:!((h:codeEditor,i:(filename:'1',fontScale:14,fontUsePx:'0',j:1,lang:c%2B%2B,selection:(endColumn:39,endLineNumber:14,positionColumn:39,positionLineNumber:14,selectionStartColumn:39,selectionStartLineNumber:14,startColumn:39,startLineNumber:14),source:'%23include+%3Ciostream%3E%0A%23include+%3Ccompare%3E%0A%23include+%3Ccstdint%3E%0A%23include+%3Cmemory%3E%0A%23include+%3Ccstring%3E%0A%0Avoid+test1(std::uint8_t*+ptr)%0A%7B%0A++++std::memset(ptr,+0,+8)%3B%0A%7D%0A%0Avoid+test2(std::uint8_t*+ptr)%0A%7B%0A++++std::fill(ptr,+ptr%2B8,+std::uint8_t%7B0%7D)%3B%0A%7D%0A%0Avoid+test3(std::uint8_t*+ptr)%0A%7B%0A++++for+(int+i+%3D+0%3B+i+!!%3D+8%3B+%2B%2Bi)%0A++++++++ptr%5Bi%5D+%3D+0%3B%0A%7D'),l:'5',n:'0',o:'C%2B%2B+source+%231',t:'0')),k:55.07177602798292,l:'4',n:'0',o:'',s:0,t:'0'),(g:!((g:!((h:compiler,i:(compiler:g111,deviceViewOpen:'1',filters:(b:'0',binary:'1',binaryObject:'1',commentOnly:'0',demangle:'0',directives:'0',execute:'0',intel:'0',libraryCode:'1',trim:'1'),flagsViewOpen:'1',fontScale:14,fontUsePx:'0',j:2,lang:c%2B%2B,libs:!((name:boost,ver:'181')),options:'-Wall+-Wextra+-pedantic+-std%3Dc%2B%2B20+-O3',selection:(endColumn:1,endLineNumber:1,positionColumn:1,positionLineNumber:1,selectionStartColumn:1,selectionStartLineNumber:1,startColumn:1,startLineNumber:1),source:1),l:'5',n:'0',o:'+x86-64+gcc+11.1+(Editor+%231)',t:'0')),k:33.33333333333333,l:'4',m:77.60097919216646,n:'0',o:'',s:0,t:'0'),(g:!((h:output,i:(editorid:1,fontScale:14,fontUsePx:'0',j:2,wrap:'1'),l:'5',n:'0',o:'Output+of+x86-64+gcc+11.1+(Compiler+%232)',t:'0')),header:(),l:'4',m:22.399020807833537,n:'0',o:'',s:0,t:'0')),k:44.92822397201707,l:'3',n:'0',o:'',t:'0')),l:'2',n:'0',o:'',t:'0')),version:4)), so the choice is mostly "what reads better".

I'm fine with changing this to use `std::fill`.

Reply:
`std::fill` is a great idea. Thanks.